app/src/main/java/com/example/eric/socials/AddEvent.java

```diff
@@ -0,0 +1,133 @@
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class AddEvent extends AppCompatActivity implements View.OnClickListener{
+
+    EditText eventName;
+    EditText date;
+    EditText description;
+    Button createSocial;
+    ImageView eventImage;
+
+    private Intent data;
+
+    FirebaseDatabase database = FirebaseDatabase.getInstance();
+    DatabaseReference myRef = database.getReference("/socials");
+    private StorageReference mStorageRef;
+
+    private static FirebaseAuth mAuth;
+    private static FirebaseUser mUser;
+
+    @Override
+    public void onClick(View v){
+        switch (v.getId()) {
+            case R.id.createEventButton:
+                mStorageRef = FirebaseStorage.getInstance().getReferenceFromUrl(getString(R.string.firebase_link));
+                final String key = myRef.child("socials").push().getKey();
+                StorageReference imageRef = mStorageRef.child(key + ".png");
+
+                imageRef.putFile(data.getData()).addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception exception) {
+                        Toast.makeText(getApplicationContext(), "need an image!", Toast.LENGTH_SHORT).show();
+                    }
+                }).addOnSuccessListener(new OnSuccessListener<UploadTask.TaskSnapshot>() {
+                    @Override
+                    public void onSuccess(UploadTask.TaskSnapshot taskSnapshot) {
+                        ArrayList<String> users = new ArrayList<String>();
+                        users.add(mUser.getUid());
+                        Social s = new Social(eventName.getText().toString(), key + ".png", mUser.getEmail() , description.getText().toString(),
+                                1, date.getText().toString(), users);
+                        myRef.child(key).setValue(s);
+                        Toast.makeText(getApplicationContext(), "Post Successful!", Toast.LENGTH_SHORT).show();
+                        Intent intent = new Intent(getApplicationContext(), FeedActivity.class);
+                        startActivity(intent);
+                    }
+                });
+                Utils.progressBar(this, getString(R.string.create_social_progressbar));
+                break;
+            default:
+        }
+    }
```

Comment:
I usually see onCreate at the top of the class. Idk how other people feel about it but that's what I've seen the most.